editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -2325,7 +2329,7 @@ bool SpriteFramesEditorPlugin::handles(Object *p_object) const {
 	if (animated_sprite_3d && *animated_sprite_3d->get_sprite_frames()) {
 		return true;
 	}
-	return p_object->is_class("SpriteFrames");
+	return !frames_editor->is_editing() && Object::cast_to<SpriteFrames>(p_object);
```

Comment:
That pattern is a bit unconventional, what makes SpriteFrames special compared to other editor plugins which handle resources?

Reply:
It handles both AnimatedSprite nodes and SpriteFrames resource. (handling nodes allows opening the editor when the resource is folded)

Reply:
This new logic is incorrect, e.g. if you have selected/edited an AnimatedSprite2D with a valid SpriteFrames and then try to edit another SpriteFrames (e.g. by double clicking in the FileSystem dock a different SpriteFrames resource file) then the SpriteFrames edited by the plugin wouldn't be changed, this new logic would refuse to edit another SpriteFrames object.

Shouldn't it be simply checking whether it's a SpriteFrames different then the currently edited one? :thinking:

Reply:
I tested that and the editor is clearing the frame before editing a new one.
Or at least does something that makes it work.
But yes, if it turns out broken, it should also check if the new SpriteFrames are different.